feat(config): 24.06 data release fixes

project-defiant opened the pull request:
## ✨ Context

24.06 data release 

<!---
Congratulations! You've made it this far!
Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your contribution.
What's the context for the changes? If the changes are related to a specific issue, please [link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) to it:
-->

## 🛠 What does this PR implement
Addresses issues found during the 24.06 data release:
 - [x] typos in airflow configuration
 - [x] grch37_to_grch_38_chain_file parameter not passed to the  LDIndexStep from LDIndexStepConfig causing step failure.

<!--- _Detailed description of the changes introduced, Give examples of the changes you've made in this pull request, include an itemized list if you can and
add diagrams or images if necessary. It'll help the reviewer_ -->

## 🙈 Missing

<!--- If there are things that are requested in the task and were not implemented, list them here -->

## 🚦 Before submitting

- [x] Do these changes cover one single feature (one change at a time)?
- [x] Did you read the [contributor guideline](https://opentargets.github.io/gentropy/development/contributing/#contributing-checklist)?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you make sure there is no commented out code in this PR?
- [x] Did you follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standards in PR title and commit messages?
- [x] Did you make sure the branch is up-to-date with the `dev` branch?
- [ ] Did you write any new necessary tests?
- [x] Did you make sure the changes pass local tests (`make test`)?
- [x] Did you make sure the changes pass pre-commit rules (e.g `poetry run pre-commit run --all-files`)?

ireneisdoomed commented on src/gentropy/ld_index.py:
```diff
@@ -57,6 +59,7 @@ def __init__(
             GnomADLDMatrix(
                 ld_matrix_template=ld_matrix_template,
                 ld_index_raw_template=ld_index_raw_template,
+                grch37_to_grch38_chain_path=grch37_to_grch38_chain_path,
```
This is OK for now, but we need to phase it out in favour of the liftover table provided by Hail, which we are already using to extract the LD matrix for a locus

project-defiant replied:
I will look over this one this week, but lets stick to the above changes for now. I have made it into an issue, feel free to modify it. https://github.com/opentargets/issues/issues/3342

ireneisdoomed replied:
Cool, thank you!

ireneisdoomed reviewed (changes requested):
There is a bug in the LD index path.
We really need to migrate this configuration to the Airflow layer sooner than later.

project-defiant commented on config/step/ot_ld_based_clumping.yaml:
```diff
@@ -1,7 +1,7 @@
   - ld_based_clumping
 
-ld_index_path: ${datasets.ld_index}
+ld_index_path: ${datasets.ld_index}/2.1.1
```
I have regenerated all gnomAD related steps.  This includes ot_ld_index and ot_variant_annotation.  The first one fetches the raw ld matrices from gnomAD for given populations and creates the ld_index in staging bucket (XX.XX).

These steps were run because there is additional ld matrix for est(Estonian population) which was missing in our original ld_index populations list. -> this resulted in slightly bigger ld_index (66GB -> 74GB). We could save it in the static assets, but I decided to add the versioning for this data source, so it exists in the staging bucket along with the original ld_index, under the 2.1.1 directory. 

Other step was run because gnomAD introduced version 4.1 for variant annotation source.  

Was there additional curation done outside of the ot_ld_index step done to the static asset you refer to ?

ireneisdoomed replied:
Thanks for the explanation! If the data is good I think it's clearer if the newer LD index replaced the one in the static_assets, or if we added the versioning there. In the end this is the location that we are going to share with partners.
In general, I'd prefer not to have the configuration for the staging phase merged into dev, which could be promoted to main.

project-defiant replied:
Sure thing! I will update the paths and shuffle the data in the static assets to match versioning.

project-defiant commented:
@ireneisdoomed I have moved the ld_index data in following fashion:
old static asset from `gs://genetics_etl_python_playground/static_assets/ld_index/` ->   `gs://genetics_etl_python_playground/static_assets/ld_index/2.1.1-exclude-est-pop/` (as this is also 2.1.1 but with missing EST population)
new ld_index generated in the 24.06 data release from `gs://genetics_etl_python_playground/output/python_etl/parquet/XX.XX/ld_index/2.1.1/` -> `gs://genetics_etl_python_playground/static_assets/ld_index/2.1.1`

Also I have updated the path in the config to match the new ld_index.

ireneisdoomed reviewed (approved):
Thanks!
